### PR TITLE
Raise error for long prompt

### PR DESCRIPTION
**Pull request by LiuXiaoxuanPKU**
This is a fix for  #113. The program hangs when the input prompt is too long because [the check ](https://github.com/vllm-project/vllm/blob/43710e8d0965d616db51639ced76605dee1bde93/vllm/core/scheduler.py#L196) will always succeed and the request will always in the waiting queue and will never be added to the running queue. 
Add a check before the if statement, if the prompt length is too long, it will raise ValueError directly.

### CODE REVIEW

**Comment by zhuohan123** [comment]
Hi Lily, please check all the possible situations [here](https://github.com/vllm-project/vllm/issues/113#issuecomment-1609806237) and make sure you can handle all of them. Thanks!

**Comment by LiuXiaoxuanPKU** [comment]
I move forward with the length check. The original code should already handle the `len(prompt) + len(generated) > limit` case, so I only deal with `model limit` and `len(prompt) > limit`.  Let me know if there are any problems, thanks!

File: vllm/config.py [hunk 1]
```diff
@@ -170,14 +170,17 @@ class SchedulerConfig:
             a single iteration.
         max_num_seqs: Maximum number of sequences to be processed in a single
             iteration.
+        max_prompt_len: Maximum length of input prompt.
```
Comment (zhuohan123) [inline vllm/config.py:173]: The actual limit should not be `max_prompt_len`. Actually, it should be `max_sequence_len` that is applied to the sum of the prompt length and the generated output length.  Can you add `max_sequence_len` check also for the sum? Thanks!

**Review by zhuohan123 — CHANGES_REQUESTED** [review]
Hey Lily, thanks for the work! Please check my comments.

File: vllm/engine/llm_engine.py [hunk 2]
```diff
@@ -284,6 +284,12 @@ def _stop_sequences(self, seq_groups: List[SequenceGroup]) -> None:
                 if stopped:
                     continue
 
+                # Check if the sequence has reached max_sequence_len.
+                if seq.get_len() == \
```
Comment (zhuohan123) [inline vllm/engine/llm_engine.py:288]: Please use `()` for line continuation: https://google.github.io/styleguide/pyguide.html#32-line-length

Comment (zhuohan123) [reply vllm/engine/llm_engine.py:288]: Nit:
```suggestion
                if seq.get_len() >= \
```

File: vllm/config.py [hunk 3]
```diff
@@ -170,14 +170,18 @@ class SchedulerConfig:
...
     """
     def __init__(
         self,
         max_num_batched_tokens: int,
         max_num_seqs: int,
+        max_sequence_len: int
     ) -> None:
         self.max_num_batched_tokens = max_num_batched_tokens
         self.max_num_seqs = max_num_seqs
+        self.max_sequence_len = max_sequence_len
```
Comment (zhuohan123) [inline vllm/config.py:184]: Nit: `max_sequence_len` -> `max_seq_len`

**Review by zhuohan123 — APPROVED** [review]
Thanks for the hard work! In general LGTM! Will merge it after the following small comments are fixed.

**Pull request merged**